include/internal/cryptlib.h

```diff
@@ -253,4 +253,31 @@ char *openssl_buf2hexstr_sep(const unsigned char *buf, long buflen, char sep);
 unsigned char *openssl_hexstr2buf_sep(const char *str, long *buflen,
                                       const char sep);
 
+static ossl_inline int ossl_ends_with_dirsep(const char *path)
+{
+    if (*path != '\0')
+        path += strlen(path) - 1;
+# if defined __VMS
+    if (*path == ']' || *path == '>' || *path == ':')
+        return 1;
+# elif defined _WIN32
+    if (*path == '\\')
+        return 1;
+# endif
+    return *path == '/';
+}
+
+static ossl_inline int ossl_is_absolute_path(const char *path)
+{
+# if defined __VMS
+    if (strchr(path, ':') != NULL)
+        return 1;
```

Comment:
On VMS, `[foo]` is an absolute path, while `[.foo]` and `[-.foo]` aren't.

File::Spec::VMS has a nearly readable regexp for it: https://metacpan.org/release/PathTools/source/lib/File/Spec/VMS.pm#L340
Essentially, what that says is that if *the first* `<` or a `[` you find *is not* followed by a `.`, a `-`, a `]`, or a `>`, it's an absolutel path.

Reply:
(`<` and `>` can be used instead of `[` and `]` as directory delimiters on VMS...  that's legacy coming all the way from TOPS-20, maybe even TOPS-10 (I've forgotten))

Reply:
fixup pushed @levitte PTAL